### ISSUE TITLE
False positive: Identical sub-expression of || and && operator

### ISSUE THREAD

**Issue by hissssst** [rrrene/credo#1025]
### Environment

```sh
$ mix credo -v
1.6.7-ref.main.2245c20+uncommittedchanges

$ elixir -v
Erlang/OTP 24 [erts-12.3.2.5] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]

Elixir 1.14.2 (compiled with Erlang/OTP 24)

$ uname -som
Linux x86_64 GNU/Linux
```

### What were you trying to do?

```elixir
defmodule X do
  import Kernel, except: [&&: 2]

  def x && x, do: true
  def _ && _, do: false
end
```

This is valid code, since is redefines the `&&` operator

### Expected outcome

No warnings, since the code above just redefines the `&&` operator. And this operator is used only in the definition signature, not in the body.

### Actual outcome

```
  Warnings - please take a look
┃
┃ [W] ↗ There are identical sub-expressions to the left and to the right of the '||' operator.
┃       lib/tria/language/ternary.ex:28 #(Tria.Language.Ternary.||)
┃ [W] ↗ There are identical sub-expressions to the left and to the right of the '&&' operator.
┃       lib/tria/language/ternary.ex:24 #(Tria.Language.Ternary.&&)
```

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by hissssst** [comment]
On this version the issue is not solved `1.7.0-rc.1-ref.master.101ac047`

```elixir
defmodule Xxx do
  @moduledoc false

  import Kernel, except: [&&: 2]

  defguard is_ternary(x) when x in ~w[yes no maybe]a

  def d && d when is_ternary(d), do: d
end
```

```
  Code Readability
┃
┃ [R] ↗ Function/macro/guard names should be written in snake_case.
┃       lib/xxx.ex:8 #(Xxx.&&)

  Warnings - please take a look
┃
┃ [W] ↗ There are identical sub-expressions to the left and to the right of the '&&' operator.
┃       lib/xxx.ex:8 #(Xxx.&&)
```

**Comment by rrrene** [comment]
> On this version the issue is not solved `1.7.0-rc.1-ref.master.101ac047`

Please not that I wrote in my previous comment that it is fixed on the branch `master` and that I included instructions how to set your Credo dependency to the branch for testing purposes.

**Comment by rrrene** [comment]
@hissssst I made a mistake the first time.

Could you check again with the Credo dep set to

```elixir
{:credo, github: "rrrene/credo"}
```

:+1:
